tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/WcfCommon.cs

```diff
@@ -92,7 +93,7 @@ internal static Scope CreateScope<TRequestContext>(TRequestContext requestContex
                 Uri requestHeadersTo = requestHeaders.To;
 
                 span.DecorateWebServerSpan(
-                    resourceName: action ?? requestHeadersTo?.LocalPath,
+                    resourceName: string.IsNullOrEmpty(action) ? UriHelpers.CleanUri(requestHeadersTo, removeScheme: true, tryRemoveIds: true) : action,
```

Comment:
In doing some testing (and learning) with WCF I found that if I did `[OperationContract (Action="")]` I'd end up getting a `string.Empty` as the `resourceName`. I'm thinking that this isn't ideal and would be better to swap to using the `URI` in the request header instead. **Is it correct that we would want to avoid empty string `resouceName`s in spans?**

Additionally, swapped from obfuscating the `httpUrl` to the actual `resourceName` - but this would go from using the `LocalPath` to the cleaned and obfuscated `AbsolutePath`. **Is it okay to use the `AbsolutePath` in place of the `LocalPath?**

Reply:
Also would it make sense to clean and obfuscate the `action` as well?

Reply:
> Is it correct that we would want to avoid empty string resouceNames in spans?

Yes, I think using `string.IsNullOrEmpty()` here is the correct thing

> Also would it make sense to clean and obfuscate the action as well?

_Maaaaaaaaybe_? 😄 It is a URL after all, but I don't _think_ it can include wildcards, so probably doesn't need obfuscating? + Bear in mind that obfuscating it now will be a breaking change for users 🙂